_P181_TempHumidity_SHT2x.ino

```diff
@@ -0,0 +1,362 @@
+
+};
+
+#endif
+
+//==============================================
+// SHT2x LIBRARY - SHT2x.cpp
+// =============================================
+SHT2X::SHT2X(uint8_t addr)
+{
+  _i2c_device_address = addr;
+  //Wire.begin();   called in ESPEasy framework
+/*
+  // Set to periodic mode
+  Wire.beginTransmission(_i2c_device_address);
+  Wire.write(0x20);   // periodic 0.5mps
+  Wire.write(0x32);   // repeatability high
+  Wire.endTransmission();
+*/
+}
+
+void SHT2X::get()
+{
+    tmp = GetTemperature();
+    hum = GetHumidity();
+    abshum = GetAbsoluteHumidity (tmp,hum);
+  //  dew = GetDewpoint(tmp,hum);
+    doVent = (float)GetDoVent(tmp,hum);
+}
+/**********************************************************
+* GetHumidity
+* Gets the current humidity from the sensor.
+*
+* @return float - The relative humidity in %RH
+* return NaN in case of error.
+**********************************************************/
+float SHT2X::GetHumidity()
+{
+  delay(50); // prevent interference from ongoing wifi transmissions  
+  double H; 
+  uint16_t val = readSensor(eRHumidityNoHoldCmd); 
+  if (val) {
+    H = (-6.0 + 125.0 / 65536.0 * (double)(val));
+    return H;
+    }
+  else
+     return NAN; 
+  }
+/**********************************************************
+* GetTemperature
+* Gets the current temperature from the sensor.
+*
+* @return float - The temperature in Deg C
+* return -100 in case of error.
+**********************************************************/
+float SHT2X::GetTemperature()
+{
+  uint16_t val = readSensor(eTempNoHoldCmd);
+  double T;
+  if (val) {
+    T = (-46.85 + 175.72 / 65536.0 * (double)(val));    // read sensor and store value in internal variable
+    return T;  
+    }
+  else 
+    return NAN;
+ }
+/**********************************************************
+* GetDewPoint
+* Calculates the dew point
+* does not read the sensor
+* @return float - The dew point in Â°C
+**********************************************************/
+/* max 4 variables per sensor :-(
+float  SHT2X::GetDewpoint(float t, float rh)
+{
+if (isnan(t) || isnan(rh) ) return NAN; 
+double H = (log10(rh)-2)/0.4343 + (17.62*t)/(243.12+t);
+double Dp = 243.12*H/(17.62-H); // this is the dew point in Celsius
+return Dp;
+}
+ */
+/**********************************************************
+* GetAbsoluteHumidity
+* Gets the absolute humidity from the sensor.
+* @return float - The abs humidity in grams/m3
+**********************************************************/
+float SHT2X::GetAbsoluteHumidity(float t, float rh)
+{
+  if (isnan(t) || isnan(rh) ) return NAN; 
+  double k1 = (17.67*t) / (t+253.5);
+  return (6.112*exp(k1) *rh*2.1674) / (273.15+t);
+}
+
+/**********************************************************
+* GetDoVent
+* Gets a recommendation to ventilate the room to prevent mold
+* @return true/false
+**********************************************************/
+
+bool SHT2X::GetDoVent(float t, float rh){
```

Comment:
this is really interesting. i wanted to use an ESP and domoticz to control my venting system, to prevent mold and stuff. i didnt know there was a compilcated formula for it to get a recommendation.

Reply:
Mold grows at certain temperature / humidity combinations. See here: http://docplayer.org/docs-images/15/53845/images/12-1.jpg. Three graphs for three different material groups (wood, concrete etc.) You want to stay below the bottommost line. The lines above it indicate how fast your mold grows.
I took that graph , transferred it to Excel, got it fitted and fetched the formula from there. So I don't think its "the" formula- but its close enough.
Maybe we should transfer it to the other temp/hum sensor plugins ?